sdk/samples/iot/paho_iot_hub_c2d_sample.c

```diff
@@ -200,10 +194,8 @@ static void receive_c2d_messages(void)
     IOT_SAMPLE_LOG(" "); // Formatting
     IOT_SAMPLE_LOG("Waiting for C2D message.\n");
 
-    if (((rc
-          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, MQTT_TIMEOUT_RECEIVE_MS))
-         != MQTTCLIENT_SUCCESS)
-        && (rc != MQTTCLIENT_TOPICNAME_TRUNCATED))
+    rc = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, MQTT_TIMEOUT_RECEIVE_MS);
+    if ((rc != MQTTCLIENT_SUCCESS) && (rc != MQTTCLIENT_TOPICNAME_TRUNCATED))
```

Comment:
Significantly cleaner, imo.

nit: Remove unnecessary brackets

```suggestion
    if (rc != MQTTCLIENT_SUCCESS && rc != MQTTCLIENT_TOPICNAME_TRUNCATED)
```

Reply:
I actually find it easier to read with the brackets here, even though they are not required.

Reply:
fan of the parens too if it means anything